drgn/helpers/linux/mm.py

```diff
@@ -291,3 +294,26 @@ def environ(task: Object) -> List[bytes]:
     env_start = mm.env_start.value_()
     env_end = mm.env_end.value_()
     return access_remote_vm(mm, env_start, env_end - env_start).split(b"\0")[:-1]
+
+
+def find_slab(prog: Program, slab_name: str) -> Optional[Object]:
+    """
+    Return the slab object corresponding to slab_name
+
+    :param slab_name: name of the slab
+    :return: ``struct kmem_cache``
+    """
+    slab_caches_addr = prog["slab_caches"].address_of_().read_()
+    for s in list_for_each_entry("struct kmem_cache", slab_caches_addr, "list"):
+        if s.name.string_().decode("utf-8") == slab_name:
```

Comment:
Rather than decoding the name of each slab cache we find, it'd be faster to encode the name we're checking against, which we only need to do once. We can also make `slab_name` accept either `str` or `bytes` for convenience: change the parameter type annotation to `slab_name: Union[str, bytes]`, then at the beginning of the function you can do:

```python
if isinstance(slab_name, str):
    slab_name = slab_name.encode()
```

Then, this check becomes simply:

```suggestion
        if s.name.string_() == slab_name:
```